src/ai/backend/agent/config/unified.py

```diff
@@ -1065,29 +1128,190 @@ class AgentUnifiedConfig(BaseConfigSchema):
+    )
+
+    model_config = ConfigDict(
+        validate_assignment=True,
+    )
+
+    def construct_unified_config(self, *, default: AgentUnifiedConfig) -> AgentUnifiedConfig:
+        agent_updates: dict[str, Any] = {}
+        if self.agent is not None:
+            agent_override_fields = self.agent.model_dump(include=self.agent.model_fields_set)
+            agent_updates["agent"] = default.agent.model_copy(update=agent_override_fields)
+        if self.container is not None:
+            container_override_fields = self.container.model_dump(
+                include=self.container.model_fields_set
+            )
+            agent_updates["container"] = default.container.model_copy(
+                update=container_override_fields
+            )
+        if self.resource is not None:
+            resource_override_fields = self.resource.model_dump(
+                include=self.resource.model_fields_set
+            )
+            agent_updates["resource"] = default.resource.model_copy(update=resource_override_fields)
+        return default.model_copy(update=agent_updates)
+
+
+class AgentUnifiedConfig(AgentGlobalConfig, AgentSpecificConfig):
+    agents: list[AgentOverrideConfig] = Field(
+        default_factory=list,
+        description=textwrap.dedent("""
+        Configuration overrides for multiple agents.
+        Use this field only to define 2 or more agents, as defining only one
+        agent using this field is redundant. Use the fields agent, container,
+        and resource to define the configuration at a global level.
+        Any field populated in the agents config will be treated as an
+        override to the global default values. Thus the global fields must still
+        be provided when defining multiple agents.
+        """),
+    )
+
     # TODO: Remove me after changing config injection logic
     model_config = ConfigDict(
         extra="allow",
+        validate_assignment=True,
     )
 
     @property
-    def agent_backend(self) -> AgentBackend:
-        return self.agent.backend
+    def agent_common(self) -> CommonAgentConfig:
+        return self.agent
 
-    @model_validator(mode="after")
-    def _validate_kubernetes_config(self) -> Self:
-        if self.agent_backend == AgentBackend.KUBERNETES:
-            if self.container.scratch_type == ScratchType.K8S_NFS and (
-                self.container.scratch_nfs_address is None
-                or self.container.scratch_nfs_options is None
-            ):
-                raise ValueError(
-                    "scratch-nfs-address and scratch-nfs-options are required for k8s-nfs"
-                )
-        return self
+    @property
+    def agent_default(self) -> OverridableAgentConfig:
+        return self.agent
+
+    @property
+    def agent_configs(self) -> Sequence[AgentUnifiedConfig]:
+        agent_configs = [agent.construct_unified_config(default=self) for agent in self.agents]
```

Comment:
When the operation seems to have some load, please remove the property.
This action seems to be called every time, so if the value is fixed, it would be good to set something like lru_cache.